lib/tbd/geo.rb

```diff
@@ -250,6 +284,8 @@ def properties(model = nil, surface = nil)
     return mismatch("model", model, cl1, mth)          unless model.is_a?(cl1)
     return mismatch("surface", surface, cl2, mth)      unless surface.is_a?(cl2)
 
+    return nil unless validate(surface)
```

Comment:
@macumber : Logs (and altogether skips) invalid OpenStudio surfaces - before populating Topolys. The vast majority of (valid) surfaces would nonetheless be derated by TBD - those _connected_ to invalid ones would be partially derated. Still need to apply this to subsurfaces. As is, the _validate_ method would also catch collinear vertices (< 10mm apart) along an edge - so considering deleting intermediate collinear points from the TBD/Topolys cloned surfaces. Work in progress ...